crates/tx-parser/src/filter.rs

```diff
@@ -47,46 +119,45 @@ pub fn derive_tx_filter_rules(params: &RollupParams) -> anyhow::Result<Vec<TxFil
         TxFilterRule::Deposit(deposit_provider),
     ])
 }
-/// Filter all the relevant [`Transaction`]s in a block based on given [`TxFilterRule`]s
-pub fn filter_relevant_txs(block: &Block, filters: &[TxFilterRule]) -> Vec<ProtocolOpTxRef> {
+
+/// Filter protocol operatios as refs from relevant [`Transaction`]s in a block based on given
+/// [`TxFilterRule`]s
+pub fn filter_protocol_op_tx_refs(
+    block: &Block,
+    filter_config: TxFilterConfig,
+) -> Vec<ProtocolOpTxRef> {
+    let filter_rules = filter_config.into_rules();
```

Comment:
The config is converted to existing rules with added variants. We can add complex logic here if needed. But this should probably suffice.